### PR TITLE
EZP-25854 fixing clean data for article

### PR DESCRIPTION
**Pull request by janit**
The content of type article has a faulty INSERT statement with values being defined as "0" where the they should be "NULL".

Details: https://jira.ez.no/browse/EZP-25854

@crevillo: Moved this here and fixed issue with missing comma. Thanks for the heads up!

### CODE REVIEW

**Comment by andrerom** [comment]
We have had these kind of proposed changes before, isn't this contradicting what legacy expected?
As in where is the assumption _"should be null"_ from?

**Comment by crevillo** [comment]
maybe similar to https://github.com/ezsystems/ezpublish-kernel/pull/1618
we had that merged but then it was reverted because bc... don't know if that bc could arise here with articles though...

**Comment by janit** [comment]
The assumption that it "should be null" comes from the error that is related to editing the content type with the current SQL.

This is the exception message that eZ Platform exports upon editing the "Article" Content Type with the included clean data dump:

```
[3/4] DriverException: An exception occurred while executing 'INSERT INTO `ezcontentclass_attribute` (`id`, `contentclass_id`, `version`, `serialized_name_list`, `serialized_description_list`, `identifier`, `category`, `placement`, `data_type_string`, `can_translate`, `is_required`, `is_information_collector`, `data_float1`, `data_float2`, `data_float3`, `data_float4`, `data_int1`, `data_int2`, `data_int3`, `data_int4`, `data_text1`, `data_text2`, `data_text3`, `data_text4`, `data_text5`, `serialized_data_text`, `is_searchable`) VALUES (:placeholder1, :placeholder2, :placeholder3, :placeholder4, :placeholder5, :placeholder6, :placeholder7, :placeholder8, :placeholder9, :placeholder10, :placeholder11, :placeholder12, :placeholder13, :placeholder14, :placeholder15, :placeholder16, :placeholder17, :placeholder18, :placeholder19, :placeholder20, :placeholder21, :placeholder22, :placeholder23, :placeholder24, :placeholder25, :placeholder26, :placeholder27)' with params [154, 2, 1, "a:1:{s:6:\"eng-GB\";s:5:\"Image\";}", "b:0;", "image", "", 7, "ezobjectrelation", 1, 0, 0, null, null, null, null, 0, "", null, null, null, null, null, null, null, "N;", 1]:

SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'data_int2' at row 1   +
```

**Comment by andrerom** [comment]
> maybe similar to #1618

That is something similar, but different, I was thinking of field type content data in regards to `NULL` use

**Comment by andrerom** [comment]
> maybe similar to #1618

I was more thinking about discussion that was made when @bdunogier / @SylvainGuittard made the initial dump, and it started adding null, as kernel uses, as opposed to `0` or `''`.

Same goes for `always-available` being now removed in the diff, these are all potential bugs in kernel that we silence with these kind of changes.

**Comment by bdunogier** [comment]
Should we start by summarizing the expectations from kernel & legacy in this regard ? I only see us going in circles on that topic unless we have a clear map of the situation.

**Comment by andrerom** [comment]
@bdunogier any pov? I'm leaning towards saying +1 and getting this to 1.3 and master, and pinging web team so they can apply it on other demos.

**Comment by bdunogier** [comment]
I'm sorry to ask that, but does it break compatibility with Legacy ?

**Comment by andrerom** [comment]
unsure

File: data/cleandata.sql [hunk 1]
```diff
@@ -53,7 +53,7 @@ INSERT INTO `ezcontentclass_attribute` (`can_translate`, `category`, `contentcla
...
+INSERT INTO `ezcontentclass_attribute` (`can_translate`, `category`, `contentclass_id`, `data_float1`, `data_float2`, `data_float3`, `data_float4`, `data_int1`, `data_int2`, `data_int3`, `data_int4`, `data_text1`, `data_text2`, `data_text3`, `data_text4`, `data_text5`, `data_type_string`, `id`, `identifier`, `is_information_collector`, `is_required`, `is_searchable`, `placement`, `serialized_data_text`, `serialized_description_list`, `serialized_name_list`, `version`) VALUES (1,'',2, NULL, NULL, NULL, NULL, 0, 51, NULL, NULL, NULL, NULL, NULL, NULL, NULL, 'ezobjectrelation', 154, 'Image', 0, 0, 1, 7, 'N;', 'a:0:{}', 'a:1:{s:6:\"eng-GB\";s:5:\"image\";}', 0);
```
Comment (andrerom) [inline data/cleandata.sql]: Actually, the discussion is not needed, then change here is doing way to much here to begin with.
Not only is "always-avaiable" info stripped, identifier is changed from `image` to `Image.`

@janit could you redo the change changing *only* what was causing issues here?

**Comment by emodric** [comment]
> I'm sorry to ask that, but does it break compatibility with Legacy ?

Well, seeing how there are other fields having the same values as changes proposed here, I would say no, but not sure 100%.

File: data/cleandata.sql (hunk 1 above)
Comment (andrerom) [inline data/cleandata.sql]: Also error message in issue does not make sense, `data_int2 ` *is* **0** here, and not `''` like error message says. Wrong line maybe?

**Comment by andrerom** [comment]
-1 This seems to be the wrong fix.

**Comment by andrerom** [comment]
Closing in favor of #1676 *(if patch there is updated to reflect discussion)*.

**Pull request closed without merging**